QueryBuilder.Test/QueryBuilder.Typed/Select.UnitTests.cs

```diff
@@ -269,5 +269,95 @@ public void CannotSelectMissingType()
                 .Join<Space, Space>(s => s.HasChildren, "spc1", "spc2")
                 .Select<Building>();
         }
+
+        [TestMethod]
+        public void CanSelectRelationship()
+        {
+            var query = QueryBuilder
+                .From<Building>()
+                .Join<Building, Floor>(s => s.HasChildren)
+                .Where<Building>(b => b.Id, ComparisonOperators.IsEqualTo, "ID")
+                .Select<Building>()
+                .Select<SpaceHasChildrenRelationship>()
+                .Select<Floor>();
+
+            var expectedQuery = $"SELECT building, spacehaschildrenrelationship, floor FROM DIGITALTWINS building JOIN floor RELATED building.hasChildren spacehaschildrenrelationship WHERE IS_OF_MODEL(building, '{Building.ModelId.UpdateVersion(1)}') AND IS_OF_MODEL(floor, '{Floor.ModelId.UpdateVersion(1)}') AND building.$dtId = 'ID'";
+
+            Assert.AreEqual(expectedQuery, query.BuildAdtQuery());
+        }
+
+        [TestMethod]
+        public void CanSelectRelationshipWithAlias()
+        {
+            var query = QueryBuilder
+                .From<Building>("b")
+                .Join<Building, Floor>(s => s.HasChildren, "b", "f", "rel")
+                .Where<Building>(b => b.Id, ComparisonOperators.IsEqualTo, "ID")
+                .Select<Building>()
+                .Select<SpaceHasChildrenRelationship>()
```

Comment:
What if someone tries to select a relationship that is not used in the query?

Reply:
That will throw an `ArgumentException` -- I'll have a unit test added to validate that.